Userland/Libraries/LibWeb/SVG/SVGPathElement.cpp

```diff
@@ -390,8 +390,33 @@ float PathDataParser::parse_fractional_constant()
 float PathDataParser::parse_number()
 {
     auto number = parse_fractional_constant();
-    if (match('e') || match('E'))
-        TODO();
+
+    if (!match('e') && !match('E'))
+        return number;
+    consume();
+
+    auto exponent_sign = parse_sign();
+
+    StringBuilder exponent_builder;
+    while (!done() && isdigit(ch()))
+        exponent_builder.append(consume());
+    VERIFY(exponent_builder.length() > 0);
```

Comment:
What happens when there's nothing after the `e` - we crash? What *should* happen?

Reply:
unfortunately this whole parser is built without any success/error reporting capabilities, so currently it asserts on unexpected input (see parse_number, parse_flag, parse_whitespace, etc), but what is likely supposed to happen is just some kind of a html parsing error